src/plugins/inputs/part.js

```diff
@@ -23,5 +23,9 @@ exports.input = function(network, chan, cmd, args) {
 		chan: chan.id
 	});
 
+	if (chan.type === "channel") {
+		this.save();
```

Comment:
Can't we put that at line 18 (and I guess moving line 21 above the block lines 16-19 then)